packages/@uppy/aws-s3-multipart/src/HTTPCommunicationQueue.ts

```diff
@@ -0,0 +1,430 @@
+import type { Meta, UppyFile } from '@uppy/utils/lib/UppyFile'
+import { pausingUploadReason, type Chunk } from './MultipartUploader.ts'
+import type AwsS3Multipart from './index.ts'
+import { throwIfAborted } from './utils.ts'
+import type { Body, UploadPartBytesResult, UploadResult } from './utils.ts'
+import type { AwsS3MultipartOptions, uploadPartBytes } from './index.ts'
+
+type RateLimitedQueue = any // untyped
```

Comment:
This is now typed since #4981.

I needed something like this for transloadit plugin to work with the new types.

```ts
export type WrapPromiseFunctionType<T extends (...args: any[]) => any> = (
  ...args: Parameters<T>
) => AbortablePromise<ReturnType<T>>
```

```ts
  #fetchWithNetworkError: WrapPromiseFunctionType<typeof fetchWithNetworkError>
```